### PR TITLE
Output pane ts conversion

### PR DESCRIPTION
**Pull request by jeremy-rifkin**

### CODE REVIEW

File: static/panes/output.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import { Toggles } from '../widgets/toggles';
+import _ from 'underscore';
+import $ from 'jquery';
```
Comment (matsjla) [inline static/panes/output.ts:27]: We don't need to explicitly import JQuery

**Review by matsjla — CHANGES_REQUESTED** [review]

File: static/panes/output.ts (hunk 1 above)
Comment (jeremy-rifkin) [reply static/panes/output.ts:27]: Thanks :P

File: static/panes/output.ts [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        const state = {
+            wrap: options.wrap,
+            ...parent,
+        };
+        this.fontScale.addState(state);
+        return state as any;
+    }
+
+    addOutputLines(result) {
+        for(const obj of (result.stdout || []).concat(result.stderr || [])) {
```
Comment (partouf) [inline static/panes/output.ts:127]: in a compiled language i would heavily question this line of code. I'm sure it doesn't matter, but can you const var this list a line before the for loop?

File: static/panes/output.ts [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+                        this.programOutput(this.normalAnsiToHtml.toHtml(obj.text));
+                    }
+                }
+            }
+        }
+        this.setCompileStatus(false);
+        this.updateTitle();
+    }
+
+    override onCompiler(compilerId: number, compiler: unknown, options: unknown, editorId: number,
```
Comment (partouf) [inline static/panes/output.ts:193]: where is this coming from exactly? i dont see it in the original .js file

Comment (jeremy-rifkin) [reply static/panes/output.ts:193]: `onCompiler` is abstract in pane.ts so we have to provide an implementation even if it the pane doesn't do anything for the event

File: static/panes/output.ts (hunk 2 above)
Comment (jeremy-rifkin) [reply static/panes/output.ts:127]: Quick clarification, are you suggesting pulling the full `(result.stdout || []).concat(result.stderr || [])` part or both `result.std___ || []` parts?

Comment (matsjla) [reply static/panes/output.ts:127]: I think it'd be a little more clear if both || []'s were pulled out

Comment (mattgodbolt) [reply static/panes/output.ts:127]: Will hold off merging until this has been replied to - I can easily imagine such code came from my original version; and we can almost certainly do better, but I wouldn't gate @jeremy-rifkin 's change on this, necessarily.

Comment (jeremy-rifkin) [reply static/panes/output.ts:127]: I'll go ahead and pull both out, @matsjla agreed it would look a little more readable

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**